### PR TITLE
Fix app breaking when canceling wallet connection

### LINKED ISSUE #1796: Closing wallet selection modal breaks the app
**To Reproduce**
1. Go to Markets.
2. Click on a pool card.
3. Click on the "Connect Wallet" button.
4. When the wallet selection modal appears, click on the close button without selecting any wallet.

**Expected behavior**
The modal closes and returns to the deposit preview modal.

**Actual behavior**
The app breaks.

### PR DESCRIPTION
**Pull request by iamvukasin**
Closes #1796

### CODE REVIEW

File: packages/tempus-client_v3/src/components/Wallet/Wallet.tsx [hunk 1]
```diff
@@ -93,8 +93,11 @@ const Wallet: FC<WalletProps> = props => {
 
   const navigate = useNavigate();
 
-  const redirectAfterConnect = useCallback(() => {
-    if (redirectTo) {
+  /**
+   * Redirects after wallet is connected, if needed.
+   */
+  useEffect(() => {
+    if (wallet && redirectTo) {
```
Comment (ShawTim) [inline packages/tempus-client_v3/src/components/Wallet/Wallet.tsx:100]: what if the wallet changed here, say switching between wallet accounts. will it trigger a redirect here?

Comment (iamvukasin) [reply packages/tempus-client_v3/src/components/Wallet/Wallet.tsx:100]: Nothing would happen. Right now, we use `Wallet` with redirect in one place and it's shown only when a user doesn't have a connected wallet. It's not future-proof though.

Comment (iamvukasin) [reply packages/tempus-client_v3/src/components/Wallet/Wallet.tsx:100]: Nothing would happen. Right now, we use `Wallet` with redirect in one place and it's shown only when a user doesn't have a connected wallet. It's not future-proof though.

Comment (ShawTim) [reply packages/tempus-client_v3/src/components/Wallet/Wallet.tsx:100]: can you add a TODO so that we can revisit this part

**Review by ShawTim — DISMISSED** [review]

**Review by ShawTim — APPROVED** [review]

**Pull request merged**